### PR TITLE
Merged Position and Vector2

### PR DESCRIPTION
**Pull request by matrefeytontias**
Behaviour of Vector2 is now entirely transparent and identical to the former Position, which has been removed. Added fancy features include operator overloading for + - * /, and all vector math functions are now reentrant.

### CODE REVIEW

File: haxepunk/math/MathUtil.hx [hunk 1]
```diff
@@ -95,8 +95,7 @@ class MathUtil
 			return;
 		}
 		point.normalize(distance);
-		object.x += point.x;
-		object.y += point.y;
+		object += point;
```
Comment (bendmorris) [inline haxepunk/math/MathUtil.hx]: This is no longer modifying `object` in place.

File: haxepunk/math/MathUtil.hx [hunk 2]
```diff
@@ -105,13 +104,11 @@ class MathUtil
 	 * @param	anchor		The anchor object.
 	 * @param	distance	The max distance object can be anchored to the anchor.
 	 */
-	public static inline function anchorTo(object:Position, anchor:Position, distance:Float = 0)
+	public static inline function anchorTo(object:Vector2, anchor:Vector2, distance:Float = 0)
 	{
-		point.x = object.x - anchor.x;
-		point.y = object.y - anchor.y;
+		point = object - anchor;
```
Comment (bendmorris) [inline haxepunk/math/MathUtil.hx]: It looks like this is allocating a new object each time. The in place modification is preferable even if it's uglier.

File: haxepunk/math/MathUtil.hx (hunk 1 above)
Comment (matrefeytontias) [reply haxepunk/math/MathUtil.hx]: Oh yeah, true. Good thing is that I can just override the = operator for assignment !

File: haxepunk/math/MathUtil.hx (hunk 2 above)
Comment (matrefeytontias) [reply haxepunk/math/MathUtil.hx]: Yeah, I'll revert that one.

File: haxepunk/math/MathUtil.hx (hunk 1 above)
Comment (matrefeytontias) [reply haxepunk/math/MathUtil.hx]: Well turns that a bug in Haxe doesn't allow me to do that, so I'll revert that as well.

**Pull request merged**